File: src/main/java/com/github/frcsty/frozenactions/actions/player/TeleportAction.java
package com.github.frcsty.frozenactions.actions.player;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.jetbrains.annotations.NotNull;

public final class TeleportAction implements Action {

    @Override
    public @NotNull String getId() {
        return "TELEPORT";
    }

    @Override
    public void run(final @NotNull ActionContext context) {
        final String[] arguments = context.getDataAsStringArray(";");
        final Location location = new Location(
                Bukkit.getWorld(arguments[0]),
                Double.parseDouble(arguments[1]),
                Double.parseDouble(arguments[2]),
                Double.parseDouble(arguments[3]),
                Float.parseFloat(arguments[4]),
                Float.parseFloat(arguments[5])
        );

        context.getPlayer().teleport(location);
    }

}

File: src/main/kotlin/com/github/frcsty/frozenactions/actions/player/TeleportAction.kt
package com.github.frcsty.frozenactions.actions.player

import com.github.frcsty.frozenactions.actions.Action
import org.bukkit.Bukkit
import org.bukkit.Location
import org.bukkit.entity.Player

object TeleportAction : Action {

    override val id = "TELEPORT"

    override fun run(player: Player, data: String) {
        val args = data.split(";")
        val location = Location(
                Bukkit.getWorld(args[0]),
                args[1].toDouble(),
                args[2].toDouble(),
                args[3].toDouble(),
                args[4].toFloat(),
                args[5].toFloat()
        )

        player.teleport(location)
    }
}